src/ploigos_step_runner/step_implementers/report/result_artifacts_archive.py

```diff
@@ -116,6 +134,38 @@ def _run_step(self):
             description='Archive of all of the step result artifacts marked for archiving.'
         )
 
+        # if destination URL specified, then upload the results archvie
```

Comment:
Would it make sense to have this block of code in its own method? I understand the goal is for the step implementer to both run the step and then upload the artifacts but what if there is a case (maybe there won't be though) where a step is run but artifacts are not uploaded anywhere?